pipeline/checksums/checksums.go

```diff
@@ -0,0 +1,64 @@
+package checksums
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/goreleaser/goreleaser/checksum"
+	"github.com/goreleaser/goreleaser/context"
+	"golang.org/x/sync/errgroup"
+)
+
+// Pipe for checksums
+type Pipe struct{}
+
+// Description of the pipe
+func (Pipe) Description() string {
+	return "Calculating checksums"
+}
+
+// Run the pipe
+func (Pipe) Run(ctx *context.Context) (err error) {
+	var g errgroup.Group
+	for _, artifact := range ctx.Artifacts {
```

Comment:
I'm only asking because I had an issue with this in my first version.

It seem to not be safe at least ref (https://golang.org/ref/spec#For_statements):

> If map entries are created during iteration, that entry may be produced during the iteration or may be skipped